javascript/javascript1/week4/get-reply.js

```diff
@@ -0,0 +1,138 @@
+// voice assistant function in javascript
+/* name should be global variable beacause the name is extracted 
+in the first call to getReply and it is displayed in the second call 
+ */
+let name;
+let todo = [];
+function getReply(inputString) {
+  inputString = inputString.trim();
+  if (inputString.search("Hello my name is") == 0) {
+    // this block of code is say hello to user and store user name in the global variable
+    let positionOfName =
+      inputString.search("Hello my name is Benjamin") +
+      "Hello my name is".length;
+    name = inputString.substring(positionOfName);
+    return "Nice to meet you " + name;
+  }
+  if (inputString.search("What is my name") == 0) {
+    // this block of code is dispaly name
+    if (name != null) return "Your name is " + name;
+    else return "Name is not defined";
+  }
+  // the following code is to add activity to "todo list"
+  if (inputString.search("Add") == 0) {
+    // now we have to find the activity
+    let activity = inputString.substring(4, inputString.search("to my todo"));
+    // length of add is 3 and one space so total 4
+    todo.push(activity);
+    return activity + "added to your todo";
+  }
+
+  // the following code is to remove from "todo list"
+  if (inputString.search("Remove") == 0) {
+    let activity = inputString.substring(7, inputString.search("from my todo"));
+    todo.pop(activity);
+    return "Removed " + activity + "from your todo";
+  }
```

Comment:
`pop` Removes the last element from an array and returns it. It doesn't take any argument.
You can use `splice`. Let's replace the text `add' and `to my todo` with an empty string `''` 
then get the index of the item and remove it using splice
```js
  if (inputString.startsWith('remove') && inputString.endsWith('from my todo')) {
    const todo = inputString.replace('add', '').replace(' to my todo', '');
    const todoIndex = todos.indexOf(todo);
    todos.splice(todoIndex, 1);
    return `${todo} removed from your todo`;
  }
```
another way using `filter` which you will learn later in JavaScript